Automattic/jetpack
# Jetpack: add VideoPress beta block scaffolding

This PR adds the scaffolding of the VideoPress (beta) block. The code lives into the `v6/` folder:

![image](https://user-images.githubusercontent.com/77539/175333955-b4e5a601-67b7-4625-8173-91466361ee57.png)


Fixes #

#### Changes proposed in this Pull Request:

* Jetpack: add VideoPress beta block scaffolding 

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:

This implementation defines the VideoPress block as a [beta extension](https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/jetpack/extensions#beta-extensions), meaning that it won't be available in production, hidden by default. To be able to enable the block, you need to:

* For the local environment, be sure that beta blocks are enabled by setting the global `JETPACK_BETA_BLOCKS` constant:
```php
define( 'JETPACK_BETA_BLOCKS', true );
```
* In the WordPress.com environment, Automatticians will be able to see beta extensions with no further configuration
* In a Jurassic Ninja site, you must go to Settings > Jetpack Constants, and enable the `JETPACK_BETA_BLOCKS` option there.
* Create/edit a post via the block editor
* Confirm the VideoPress block is available 
* Confirm it shows the `(beta)` label
* Save the post
* Confirm the block renders in the front-end (you should see a log in the console)

<img width="338" alt="Screen Shot 2022-06-22 at 7 03 36 PM" src="https://user-images.githubusercontent.com/77539/175160163-9baeb6de-d4a2-42ca-983c-bde9e636128c.png">
<img width="369" alt="Screen Shot 2022-06-22 at 7 03 14 PM" src="https://user-images.githubusercontent.com/77539/175160169-07e47d50-3307-4776-8ae8-754eb3f78631.png">

<img width="688" alt="image" src="https://user-images.githubusercontent.com/77539/175160113-ca3387dc-b310-43eb-b696-19df10f6839c.png">

<img width="732" alt="image" src="https://user-images.githubusercontent.com/77539/175160146-354c286f-89fe-4d24-a8ca-3f62e51998b1.png">

<img width="311" alt="Screen Shot 2022-06-23 at 1 00 10 PM" src="https://user-images.githubusercontent.com/77539/175343818-8f7e5d09-989b-4beb-b2c9-2e94069971bf.png">

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202490094773323
